wgengine/userspace.go

```diff
@@ -314,6 +321,31 @@ func echoRespondToAll(p *packet.ParsedPacket, t *tstun.TUN) filter.Response {
 	return filter.Accept
 }
 
+// handleLocalPackets inspects packets coming from the local network
+// stack, and intercepts any packets that should be handled by
+// tailscaled directly. Other packets are allowed to proceed into the
+// main ACL filter.
+func (e *userspaceEngine) handleLocalPackets(p *packet.ParsedPacket, t *tstun.TUN) filter.Response {
+	if e.useTailscaleDNS {
+		if verdict := e.handleDNS(p, t); verdict == filter.Drop {
+			// local DNS handled the packet.
+			return filter.Drop
+		}
+	}
+
+	// TODO: atomic value
+	localAddrs := e.localAddrs.Load().(map[packet.IP]bool)
```

Comment:
be defensive here and use `, ok`? I guess crashing if `Load` returns `nil` is a feature? Still, it's a bit scary.

Reply:
It's a feature in that I initialize e.localAddrs to an empty map explicitly at construction, so Load should never be able to read a nil out.

But, switched to `, ok` and an unexpected log for defensiveness.